pallets/parachain-staking/src/auto_compound.rs

```diff
@@ -205,7 +209,7 @@ where
 
 		if !auto_compound.is_zero() {
 			ensure!(
-				Self::get_storage(&candidate).len()
+				Self::get_auto_compounding_delegation_count(&candidate) as u32
```

Comment:
We end up calling `get_storage(&candidate)` below in probably the most common code path, so this might actually make perf worse in that case.

Reply:
Actually, maybe not. I think the `decode_len` impl can be incredibly cheap -- it just needs to process the first few bytes to see the length of the array.

Reply:
I had the same opinion. Let's see the benchmarking results.